crates/matrix-sdk/src/room/timeline/mod.rs

```diff
@@ -443,6 +347,136 @@ impl Timeline {
     }
 }
 
+/// Builder that allows creating and configuring various parts of a
+/// [`Timeline`].
+#[must_use]
+#[derive(Debug)]
+pub(crate) struct TimelineBuilder {
+    room: room::Common,
+    prev_token: Option<String>,
+    events: Vec<SyncTimelineEvent>,
+    track_fully_read: bool,
+}
+
+impl TimelineBuilder {
+    pub(super) fn new(room: &room::Common) -> Self {
+        Self {
+            room: room.clone(),
+            prev_token: None,
+            events: Vec::default(),
+            track_fully_read: false,
+        }
+    }
```

Comment:
Instead of making this public outside the timeline module, can we have `Timeline::builder()` as `pub(super)` / `pub(crate)` (which can internally call this)?